src/plugins/data_source_management/public/components/edit_data_source/components/edit_form/edit_data_source_form.tsx

```diff
@@ -242,8 +261,22 @@ export class EditDataSourceForm extends React.Component<EditDataSourceProps, Edi
     this.setState({ showUpdatePasswordModal: true });
   };
 
-  updatePassword = (passwords: UpdatePasswordFormType) => {
-    // TODO: update password when API is ready
+  /* Update password */
+  updatePassword = (password: string) => {
+    const { title, description, auth } = this.props.existingDataSource;
+    const updateAttributes: DataSourceAttributes = {
+      title,
+      description,
+      endpoint: undefined,
+      auth: {
+        type: auth.type,
+        credentials: {
+          username: auth.credentials ? auth.credentials.username : '',
```

Comment:
Shouldn't we allow empty user name?

Reply:
This is to avoid es lint error .. there is no way for user to update with empty user name